### PR TITLE
[FIX] core: remove SQL constraints upon ir.model.constraint removal

### PR DESCRIPTION
**Pull request by aj-fuentes**
Otherwise we leave the constraints in the table. Common source of upgrade issues.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/base/models/ir_model.py [hunk 1]
```diff
@@ -1518,13 +1518,7 @@ class IrModelConstraint(models.Model):
...
     ]
 
-    def _module_data_uninstall(self):
-        """
-        Delete PostgreSQL foreign keys and constraints tracked by this model.
-        """
-        if not self.env.is_system():
-            raise AccessError(_('Administrator access is required to uninstall a module'))
-
+    def unlink(self):
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_model.py:1521]: I suggest to check access rights before actually removing SQL constraints.
```suggestion
    def unlink(self):
        self.check_access_rights('unlink')
        self.check_access_rule('unlink')
```

File: odoo/addons/base/models/ir_model.py [hunk 2]
```diff
@@ -2199,7 +2193,6 @@ def delete(records):
 
         # remove constraints
         constraints = self.env['ir.model.constraint'].search([('module', 'in', modules.ids)])
-        constraints._module_data_uninstall()
```
Comment (rco-odoo) [inline odoo/addons/base/models/ir_model.py:2202]: Looks like `constraints` is no longer useful, so you can remove the first line as well.

**Review by rco-odoo — COMMENTED** [review]
Makes sense to me.  I don't remember why we had this double logic for removing constraints and `ir.model.constraint` records.

**Comment by aj-fuentes** [comment]
@rco-odoo all comments addressed, thanks! Could you review this again please?

**Review by KangOl — APPROVED** [review]

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**